app/assets/javascripts/client/common_create/CreateService.js

```diff
@@ -80,5 +99,103 @@
             service.emitError('Could not save assessment');
           });
     };
+
+    this.assignAndSaveInventory = function(inventory) {
+      if (inventory.message === null || inventory.message === '') {
+        this.alertError = true;
+        return;
+      }
+
+      if ($window.confirm('Are you sure you want to start the inventory and invite all your participants?')) {
+        this.alertError = false;
+        this.saveInventory(inventory, true)
+            .then(function() {
+              $location.path('/inventories');
+            });
+      }
+    };
+
+    this.saveInventory = function(inventory, assign) {
+      if (inventory.name === '') {
+        this.emitError('Inventory needs a name!');
+        return;
+      }
+
+      inventory.district_id = this.district.id;
+      service.toggleSavingState();
+      inventory.deadline = moment($('#due-date').val(), 'MM/DD/YYYY').toISOString();
+
+      if (assign) {
+        inventory.assign = true;
+      }
+
+      return Inventory.save({inventory_id: inventory.id}, {inventory: inventory})
+          .$promise
+          .then(function() {
+            service.toggleSavingState();
+            service.emitSuccess('Inventory Saved!');
+          }, function() {
+            service.toggleSavingState();
+            service.emitError('Could not save inventory');
+          });
+    };
+
+    this.save = function(entity) {
+      if(service.context === 'assessment') {
+        this.saveAssessment(entity);
+      } else if(service.context === 'inventory') {
+        this.saveInventory(entity);
+      }
+    };
+
+    this.loadParticipants = function() {
```

Comment:
This method and the ones below might be clearer in terms of readibility if broken down into components, much like `save` does it.
```javascript
this.loadParticipants = function () {
  if(this.isAssessmentContext()) {
    return loadAssessmentParticipants();
  } else if(this.isInventoryContext()) {
    return loadInventoryParticipants();
  }
}

this.loadAssessmentParticipants = function () {
  return Participant.query({assessment_id: $stateParams.id});
}

this.loadInventoryParticipants = function () {
  return InventoryParticipant.query({inventory_id: $stateParams.id});
}
```

Using these utilities might also be beneficial:
```javascript
this.isAssessmentContext = function () { return service.context === 'assessment'; };
this.isInventoryContext = function () { return service.context === 'inventory'; };
```